store/src/java/com/zimbra/cs/datasource/imap/ImapFolderSync.java

```diff
@@ -431,10 +444,35 @@ private void syncFlags(long lastUid) throws ServiceException, IOException {
      * let it be handled when the originating folder is processed.
      */
     private void pushChanges(MessageChanges changes) throws ServiceException, IOException {
+        pushChangesForDraft(changes, null);
+    }
+
+    /**
+     * Message has been modified in either of below ways
+     * ZBUG-2361: Add condition to identify if draft content has been modified
+     * 
+     * 1. Message flags changed (UPDATED)
+     * 2. New message added to this folder (ADDED)
+     * 3. Message moved from this folder to another (MOVED)
+     * 4. Message moved to this folder from another
+     * 5. Local Draft message modified or it's Attachment changed
+     * 
+     * @param changes
+     * @param folderName
+     * @throws ServiceException
+     * @throws IOException
+     */
+    private void pushChangesForDraft(MessageChanges changes, String folderName) throws ServiceException, IOException {
         localFolder.debug("Pushing changes: %s", changes);
         for (MessageChange change : changes.getChanges()) {
             clearError(change.getItemId());
-            if (change.isAdded()) {
+            if ((change.isModified() || change.isUpdated()) && DRAFTS.equalsIgnoreCase(folderName)
+                    && deleteMessage(change.getTracker().getUid())) {
```

Comment:
whats the purpose of `deleteMessage` in the if check, is it actually deleting message?

Reply:
yes this actually deletes the message from Remote IMAP and then pushes the new modified draft to remote in next step